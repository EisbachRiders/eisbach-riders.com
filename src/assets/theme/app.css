html {
  height: 100%;
}

body {
  margin: 0;
  padding: 0;
  font-family: 'Raleway', sans-serif;
  background-color: #ffffff;
  overflow-x: hidden;
}

a {
  text-decoration: none;
}
