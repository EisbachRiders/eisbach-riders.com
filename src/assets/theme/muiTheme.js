import { createMuiTheme } from '@material-ui/core/styles'
import './App.css'
require('typeface-open-sans')
require('typeface-contrail-one')

const status = {
  white: '#ffffff',
  offWhite: '#F7F5F5',
  black: '#0b0c10',
  blue: '#1f2833',
  grey: '#c5c6c7',
  greyDk: '#404040',
  teal: '#66fcf1',
  tealDk: '#45a29e',
  green: '#66fca6',
  yellow: '#fcf166',
  red: '#e25b65',
}

const theme = createMuiTheme({
  palette: {
    type: 'light',
    primary: {
      light: status.teal,
      main: status.teal,
      dark: status.tealDk,
    },
    secondary: {
      light: status.blue,
      main: status.blue,
      dark: status.blue,
    },
    error: { main: status.red },
  },
  status,
  typography: {
    fontFamily: "'Open Sans', sans-serif",
    fontSize: 14,
  },
  overrides: {
    MuiTypography: {
      colorPrimary: {
        color: status.greyDk,
      },
      colorSecondary: {
        color: status.grey,
      },
      colorTextSecondary: {
        color: status.white,
      },
    },
  },
})

export default theme
