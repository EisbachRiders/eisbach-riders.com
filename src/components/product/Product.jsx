import React, { useState } from 'react'
import { connect } from 'react-redux'
import { useTranslation } from 'react-i18next'
import { makeStyles } from '@material-ui/styles'
import Typography from '@material-ui/core/Typography'
import classnames from 'classnames'
import Button from '@material-ui/core/Button'
import ProductAttribute from './ProductAttribute'
import ProductImg from './ProductImg'
import ProductDescription from './ProductDescription'
import Counter from '../counter/Counter'
import Container from '../ui/Container'

const mapStateToProps = ({ product, lng, cart }) => {
  return { product, lng, cart }
}

const mapDispatchToProps = dispatch => {
  return {
    addToCart: cart => dispatch({ type: `CART`, cart }),
  }
}

const useStyles = makeStyles(theme => ({
  container: {
    display: 'flex',
    justifyContent: 'space-between',
  },
  imgContainer: {
    flexBasis: '50%',
  },
  name: {
    paddingBottom: 10,
    fontWeight: 300,
    textAlign: 'left',
  },
  contentContainer: {
    flexBasis: '40%',
  },
  priceContainer: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 30,
  },
  price: {
    display: 'flex',
    justifyContent: 'space-between',
    paddingBottom: 10,
  },
  red: {
    color: theme.status.red,
  },
  divider: {
    marginBottom: 60,
    marginTop: 60,
    borderBottom: `1px solid ${theme.status.grey}`,
  },
}))

function Product({ product, cart, addToCart }) {
  const { t } = useTranslation()
  const classes = useStyles()
  const [quantity, setQuantity] = useState(1)
  const [values, setValues] = useState({
    attr1: '',
    attr2: '',
    attr3: '',
  })

  const handleRemove = () => {
    if (quantity > 0) {
      setQuantity(quantity - 1)
    }
  }

  const handleAdd = () => {
    if (quantity < 15 || quantity < product.stock_quantity) {
      setQuantity(quantity + 1)
    }
  }

  function handleChange(event) {
    setValues(oldValues => ({
      ...oldValues,
      [event.target.name]: event.target.value,
    }))
  }

  const handleCart = () => {
    let productInfo = {
      id: product.id,
      attr1: values.attr1,
      attr2: values.attr2,
      attr3: values.attr3,
    }
    const newCart = []
    if (cart.length === 0) {
      newCart.push({ quantity, productInfo })
    } else {
      cart.forEach(item => {
        if (item.product.id !== product.id) {
          newCart.push(item)
        }
      })
      newCart.push({ quantity, productInfo })
    }
    addToCart(newCart)
  }

  return (
    <Container>
      {Object.keys(product).length !== 0 && (
        <div>
          <div className={classes.container}>
            <div className={classes.imgContainer}>
              <ProductImg images={product.images} />
            </div>
            <div className={classes.contentContainer}>
              <Typography
                variant="h6"
                variantMapping="h1"
                className={classes.name}
              >
                {product.name}
              </Typography>
              <div
                dangerouslySetInnerHTML={{ __html: product.short_description }}
                className={classes.text}
              />
              <div className={classes.priceContainer}>
                <Typography variant="h6" className={classes.price}>
                  {`â‚¬${product.price}`}
                </Typography>
                {!product.in_stock && (
                  <Typography
                    className={classnames(classes.price, classes.red)}
                  >
                    {t('products.outOfStock')}
                  </Typography>
                )}
              </div>
              {product.attributes.length !== 0 &&
                product.attributes.map((elem, idx) => (
                  <ProductAttribute
                    selected={`values.attr${idx + 1}`}
                    name={`attr${idx + 1}`}
                    onChange={handleChange}
                    attribute={elem}
                    key={`attribute_${elem.name}`}
                  />
                ))}
              <div className={classes.priceContainer}>
                <Counter
                  handleRemove={handleRemove}
                  handleAdd={handleAdd}
                  value={quantity}
                />
                <Button
                  variant="contained"
                  color="primary"
                  onClick={handleCart}
                >
                  {t('products.addToCart')}
                </Button>
              </div>
            </div>
          </div>
          <div className={classes.divider} />
          <ProductDescription product={product} />
        </div>
      )}
    </Container>
  )
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Product)
