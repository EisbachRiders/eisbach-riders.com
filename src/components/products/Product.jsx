import React from "react"
import { useTranslation } from "react-i18next"
import clsx from "clsx"
import { makeStyles } from "@material-ui/core/styles"
import Typography from "@material-ui/core/Typography"
import Button from "@material-ui/core/Button"
import Container from "../ui/Container"
import Colors from "./Colors"

const useStyles = makeStyles((theme) => ({
  container: {
    marginBottom: 150,
  },
  flexContainer: {
    flexBasis: "100%",
    [theme.breakpoints.up("md")]: {
      flexBasis: "60%",
    },
  },
  flexContainer2: {
    flexBasis: "100%",
    [theme.breakpoints.up("md")]: {
      flexBasis: "38%",
    },
  },
  imgContainer: {
    display: "flex",
    flexWrap: "wrap",
    alignItems: "flex-start",
  },
  img: {
    width: "50%",
  },
  imgFull: {
    width: "100%",
  },
  img3: {
    width: "33%",
  },
  header: {
    fontFamily: "secondary",
    fontSize: 32,
    marginBottom: 30,
  },
  subheader: {
    textTransform: "capitalize",
    fontSize: 18,
    paddingRight: 15,
  },
  attributeContainer: {
    display: "flex",
    alignItems: "center",
  },
  text: {
    marginBottom: 60,
  },
  divider: {
    borderBottom: `1px solid ${theme.color.gray}`,
    marginTop: 30,
    marginBottom: 30,
  },
  buttonContainer: {
    textAlign: "center",
  },
}))

function Product({ product }) {
  const { t } = useTranslation()
  const classes = useStyles()

  let colors =
    product === undefined
      ? []
      : product.paColors.edges.map((elem) => elem.node.name)
  let sizes =
    product === undefined
      ? []
      : product.paSizes.edges.map((elem) => elem.node.name)
  let plugs =
    product === undefined
      ? []
      : product.paFinPlugs.edges.map((elem) => elem.node.name)
  let waterTemps =
    product === undefined
      ? []
      : product.paWaterTemps.edges.map((elem) => elem.node.name)
  let amounts =
    product === undefined
      ? []
      : product.paAmounts.edges.map((elem) => elem.node.name)

  return (
    <Container justifyContent="spaceBetween" className={classes.container}>
      <div className={clsx(classes.flexContainer, classes.imgContainer)}>
        <img
          src={product.image.sourceUrl}
          alt={product.image.slug}
          className={
            product.galleryImages.edges.length === 0
              ? classes.imgFull
              : classes.img
          }
        />
        {product.galleryImages.edges.length > 0 && (
          <>
            <img
              src={product.galleryImages.edges[0].node.sourceUrl}
              alt={product.galleryImages.edges[0].node.slug}
              className={classes.img}
            />
            {product.galleryImages.edges.map((elem, idx) => {
              if (idx !== 0) {
                return (
                  <img
                    key={`img${idx}`}
                    src={elem.node.sourceUrl}
                    alt={elem.node.slug}
                    className={classes.img3}
                  />
                )
              }
            })}
          </>
        )}
      </div>

      <div className={classes.flexContainer2}>
        <Typography variant="h6" component="h1" className={classes.header}>
          {product.name}
        </Typography>

        {colors.length > 0 && (
          <div className={classes.attributeContainer}>
            <p className={classes.subheader}>{t("product.colors")}:</p>
            <Colors colors={colors} variant="large" />
          </div>
        )}
        {sizes.length > 0 && (
          <div className={classes.attributeContainer}>
            <p className={classes.subheader}>{t("product.sizes")}:</p>
            {sizes.map((elem) => (
              <p className={classes.subheader} key={`size_${elem}`}>
                {elem}
              </p>
            ))}
          </div>
        )}
        {plugs.length > 0 && (
          <div className={classes.attributeContainer}>
            <p className={classes.subheader}>{t("product.plugs")}:</p>
            {plugs.map((elem) => (
              <p className={classes.subheader} key={`plug_${elem}`}>
                {elem}
              </p>
            ))}
          </div>
        )}
        {amounts.length > 0 && (
          <div className={classes.attributeContainer}>
            <p className={classes.subheader}>{t("product.amounts")}:</p>
            {amounts.map((elem) => (
              <p className={classes.subheader} key={`amounts_${elem}`}>
                {elem}
              </p>
            ))}
          </div>
        )}
        {waterTemps.length > 0 && (
          <div className={classes.attributeContainer}>
            <p className={classes.subheader}>{t("product.waterTemps")}:</p>
            {waterTemps.map((elem) => (
              <p className={classes.subheader} key={`temps_${elem}`}>
                {elem}
              </p>
            ))}
          </div>
        )}

        <div className={classes.divider}></div>

        <p className={classes.subheader}>{t("product.features")}</p>
        <div
          dangerouslySetInnerHTML={{ __html: product.shortDescription }}
          className={classes.text}
        />

        <div className={classes.buttonContainer}>
          <Button variant="contained" color="primary" href={product.link}>
            {t("product.buy")}
          </Button>
        </div>
      </div>
    </Container>
  )
}

export default Product
