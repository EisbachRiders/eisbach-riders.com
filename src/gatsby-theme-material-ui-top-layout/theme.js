import { createMuiTheme } from "@material-ui/core/styles"
import "../app.css"
import "../../static/fonts/fonts.css"

// fin: "#00d7a2",
// finDk: "#00be8f",
// finLt: "#e2f9ef",
// greyLt2: "#eeeff0",
// greyLt: "#d5d9db",
// grey: "#818C93",
// greyBlue: "#f3f8fc",
// greenLt1: "#6dff97",
// greenLt2: "#00cf3c",
// cream: "#f7f5ee",

const color = {
  // for shop colors
  blue: "#004AB5",
  green: "#00942b",
  yellow: "#fce566",
  orange: "#ff8000",
  pink: "#ff0080",
  red: "#FF0012",
  redLt: "#ff4e5b",
  // theme colors
  white: "#ffffff",
  black: "#000000",
  almostBlack: "#363738", // typography
  textLight: "#838588",
  gray: "#ebedee",
  primaryBlue: "#0062f0",
  primaryBlueLt: "#3f8eff",
  gold: "#E6C55E",
  goldLt: "#f0dda0",
}

const theme = createMuiTheme({
  palette: {
    type: "light",
    primary: {
      light: color.goldLt,
      main: color.gold,
      dark: color.goldLt,
    },
    secondary: {
      light: color.primaryBlueLt,
      main: color.primaryBlue,
      dark: color.primaryBlueLt,
    },
    error: { main: color.red },
  },
  color,
  typography: {
    fontFamily: "'Raleway', sans-serif",
    h1: {
      letterSpacing: 2,
    },
    h2: {
      letterSpacing: 2,
    },
    h3: {
      letterSpacing: 2,
    },
    h4: {
      letterSpacing: 2,
    },
    h5: {
      letterSpacing: 2,
    },
    body1: {
      letterSpacing: 1.5,
      fontSize: 14,
    },
    body2: {
      letterSpacing: 1.5,
      fontSize: 12,
    },
    button: {
      fontFamily: "'Raleway', sans-serif",
      letterSpacing: 3,
    },
  },
  overrides: {
    MuiButton: {
      root: {
        padding: "9px 27px",
        borderRadius: 0,
      },
      contained: {
        boxShadow: "none",
      },
    },
  },
})

export default theme
