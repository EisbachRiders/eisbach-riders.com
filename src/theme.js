import { createTheme } from "@material-ui/core/styles"
import "./css/app.css"

// fin: "#00d7a2",
// finDk: "#00be8f",
// finLt: "#e2f9ef",
// grayLt2: "#eeeff0",
// grayLt: "#d5d9db",
// gray: "#818C93",
// grayBlue: "#f3f8fc",
// greenLt1: "#6dff97",
// greenLt2: "#00cf3c",
// cream: "#f7f5ee",

const color = {
  // for shop colors
  blue: "#004AB5",
  green: "#00942b",
  yellow: "#fce566",
  orange: "#ff8000",
  pink: "#ff0080",
  red: "#FF0012",
  redLt: "#ff4e5b",
  seaGlass: "#5F5850",
  // theme colors
  white: "#ffffff",
  black: "#0B0C10",
  blackLt: "#1F2833",
  gray: "#C5C6C7",
  // teal: "#66FCF1",
  teal: "#00d7a2",
  tealDk: "#45A29E",
  test: "#F7F680",
  cream: "#F1F0EB",
  grayLt: "#f7f7f9",
  blueGray: "#cbd3cf",
  steel: "#505D72",
}

const theme = createTheme({
  palette: {
    type: "light",
    primary: {
      light: color.tealDk,
      main: color.teal,
      dark: color.tealDk,
    },
    secondary: {
      light: "#9d9fa0",
      main: color.gray,
      dark: "#9d9fa0",
    },
    error: { main: color.red },
  },
  color,
  typography: {
    fontFamily: ["Raleway", "sans-serif"].join(","),
  },
  components: {
    MuiButton: {
      styleOverrides: {
        root: {
          padding: "9px 27px",
          borderRadius: 0,
        },
        contained: {
          boxShadow: "none",
        },
      },
    },
  },
})

export default theme
