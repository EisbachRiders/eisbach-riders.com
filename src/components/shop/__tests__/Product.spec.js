import React from 'react'
import { render } from 'react-testing-library'
import Product from '../Product'
import { ThemeProvider } from '@material-ui/styles'
import theme from '../../../assets/theme/muiTheme'

describe('Product', () => {
  const mockProps = {
    t: () => '',
  }
  it('renders correctly', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Product {...mockProps} />
      </ThemeProvider>
    )
    expect(component).toMatchSnapshot()
  })
})
