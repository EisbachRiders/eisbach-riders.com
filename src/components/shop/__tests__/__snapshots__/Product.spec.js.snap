// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Hook-root-go6127"
      >
        <img
          class="Hook-img-cai670"
          src=""
        />
        <div
          class="Hook-descriptionContainer-1tsden"
        >
          <p
            class="MuiTypography-root-1 MuiTypography-body1-10"
          />
          <div
            class="Hook-fullWidth-t3yivt"
          >
            <div
              class="Hook-priceContainer-11bofz3"
            >
              <h6
                class="MuiTypography-root-1 MuiTypography-h6-18 Hook-price-6knxjf"
              >
                €undefined
              </h6>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Hook-root-go6127"
    >
      <img
        class="Hook-img-cai670"
        src=""
      />
      <div
        class="Hook-descriptionContainer-1tsden"
      >
        <p
          class="MuiTypography-root-1 MuiTypography-body1-10"
        />
        <div
          class="Hook-fullWidth-t3yivt"
        >
          <div
            class="Hook-priceContainer-11bofz3"
          >
            <h6
              class="MuiTypography-root-1 MuiTypography-h6-18 Hook-price-6knxjf"
            >
              €undefined
            </h6>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
